connector-email/src/main/java/com/redhat/cloud/notifications/connector/email/processors/templates/TemplateProcessor.java

```diff
@@ -0,0 +1,79 @@
+package com.redhat.cloud.notifications.connector.email.processors.templates;
+
+import com.redhat.cloud.notifications.connector.email.constants.Properties;
+import com.redhat.cloud.notifications.connector.email.model.Environment;
+import com.redhat.cloud.notifications.ingress.ActionOut;
+import io.quarkus.qute.Engine;
+import io.quarkus.qute.TemplateInstance;
+import io.quarkus.scheduler.Scheduled;
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+@ApplicationScoped
+public class TemplateProcessor implements Processor {
```

Comment:
I'm not sure whether the templates rendering should be done in the connector or in the engine. What you did here makes perfect sense for the instant emails, but daily ones might be an issue.

The main purpose of all connectors is to isolate as much as possible the engine from connections to external services (internal like BOP/IT/RBAC or external) because these connections:
- are often slow
- may fail

Rendering templates in the engine and sending the result of the rendering to the connectors wouldn't go against the initial intent of the connectors. However, the payload size might be an issue, unless it is compressed by the engine and decompressed by the connector.

I'm kind of thinking out loud here. Still hesitating :smiley:

Reply:
I gave this topic another thought and I think the connector should be "subscription type"-agnostic ideally. This would mean that templates rendering stays in the `engine`. I may have said otherwise a few weeks ago and I apologize if I did change my mind (I'm not sure :sweat_smile:).

Reply:
Kafka supports compression out of the box, so we could compress all messages going through the `platform.notifications.tocamel` topic pretty easily.

Reply:
https://docs.confluent.io/platform/current/installation/configuration/producer-configs.html#compression-type
https://camel.apache.org/components/3.21.x/kafka-component.html#_endpoint_query_option_compressionCodec

Reply:
`snappy` seems to be a popular algorithm because it doesn't slow down the processing too much (unlike `gzip`) while still compressing the data in a decent way.

This may work in `applications.properties` (untested):
```
mp.messaging.outgoing.tocamel.compression.type=snappy
```

I'm not sure of the config in Camel to decompress the data using the correct algorithm. The doc only mentions properties for the producer and not the consumer.

Reply:
`snappy` seems to be a popular algorithm because it doesn't slow down the processing too much (unlike `gzip`) while still compressing the data in a decent way.

This may work in `applications.properties` (untested):
```
mp.messaging.outgoing.tocamel.compression.type=snappy
```

I'm not sure of the config in Camel to decompress the data using the correct algorithm. The doc only mentions properties for the producer and not the consumer.

Reply:
Perfect, I'll give this info a look! This might require changes in all the connectors, though? What if we leave the compression thing for a V2? Maybe we could see how "bad" it performs in a RAW manner, and then apply compression in a separate PR along with the rest of the connectors.

Reply:
I suspect this will work automagically by simpling adding the property I mentioned. I haven't checked but it is possible that a "compression" header is added to compressed Kafka messages so that the consumer can determine which algorithm to use while uncompressing. If it is indeed that simple, let's include it in V1. If not, we can wait for V2.

Reply:
> `snappy` seems to be a popular algorithm because it doesn't slow down the processing too much (unlike `gzip`) while still compressing the data in a decent way.
> 
> This may work in `applications.properties` (untested):
> 
> ```
> mp.messaging.outgoing.tocamel.compression.type=snappy
> ```
> 
> I'm not sure of the config in Camel to decompress the data using the correct algorithm. The doc only mentions properties for the producer and not the consumer.

Did you try the property?

Reply:
Yes, I have tested it with a local Kafka instance, in order to be able to see the logs of the messages to see if compression was being applied: it was.

I am confused though because I am connecting with a consumer and with the connector itself to the topic, and I can read the messages in "plain text" directly. I thought that I would need to specify parameters somewhere so that I don't simply read garbage, but it seems like consumers are able to automatically handle that?

Reply:
Awesome, thanks for testing this!